src/PostgREST/Error.hs

```diff
@@ -52,6 +52,16 @@ class (JSON.ToJSON a) => PgrstError a where
 
 
 
+data ErrorCodePrefix
+  = ApiRequestErrorPrefix Text
+  | HasqlErrorPrefix      Text
+  | GeneralErrorPrefix    Text
+
+instance JSON.ToJSON ErrorCodePrefix where
+  toJSON (GeneralErrorPrefix    c) = JSON.toJSON ("PGRST0" <> c)
+  toJSON (ApiRequestErrorPrefix c) = JSON.toJSON ("PGRST1" <> c)
+  toJSON (HasqlErrorPrefix      c) = JSON.toJSON ("PGRST2" <> c)
```

Comment:
I found convenient to add a prefix for each group of errors that are defined in the code. This is to keep an order in the file and to identify the groups more easily. If no prefix was present, then the sequence of error codes would "jump" from group to group; e. g. if the last `Error` has a `PGRST30` code, and the last `ApiRequestError` has a `PGRST10` then the next `ApiRequestError` added would be `PGRST31`. 

The `GerenalErrorPrefix` corresponds to the `Error` data type and has a prefix of `0` due to being the errors that are not "grouped" by default and has `ApiRequestError` and `PgError` (Hasql errors) as constructors.

Reply:
I think that might be useful and it's similar to how pg classifies their errors: https://www.postgresql.org/docs/13/errcodes-appendix.html#ERRCODES-TABLE

So one group could be for JWT related errors, another for API errors, another for connection errors and so on.